edge-hub/watchdog/src/child.rs

```diff
@@ -100,13 +104,24 @@ pub fn run(
         let mut child_process = ChildProcess::new(name, child);
 
         while child_process.is_running() && !should_shutdown.load(Ordering::Relaxed) {
-            thread::sleep(PROCESS_POLL_INTERVAL_SECS);
+            sleep(PROCESS_POLL_INTERVAL)?;
         }
 
         // tell the threads to shut down their child process
         should_shutdown.store(true, Ordering::Relaxed);
 
-        child_process.shutdown_if_running();
+        child_process.shutdown_if_running()?;
+        Ok(())
     });
     Ok(handle)
 }
+
+// We can use `thread::sleep()` instead when this issue is resolved:
+// https://github.com/rust-lang/rust/issues/95661
+fn sleep(duration: Duration) -> Result<()> {
+    Command::new("sleep")
+        .arg(duration.as_secs().to_string())
+        .output()?;
+
+    Ok(())
+}
```

Comment:
We could only do this for debian 10. Thoughts?

Reply:
My concern with this is a possible performance penalty. Did we try other versions of sleep, tokio or futures?

Reply:
We considered introducing tokio runtime but thought it was better to put this in temporarily for now then switch back to thread::sleep when rust bug was fixed. However we recently found that we are removing broker from 1.3 codebase, which means the whole watchdog is disappearing.